ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -622,13 +636,27 @@ private PutResult run_put_v2(NamedBlobRecord record, NamedBlobState state, short
       } else {
         statement.setTimestamp(5, null);
       }
-      statement.setLong(6, buildVersion());
+      final long newVersion = buildVersion();
+      record.setVersion(newVersion);
+      statement.setLong(6, newVersion);
       statement.setInt(7, state.ordinal());
       statement.executeUpdate();
     }
     return new PutResult(record);
   }
 
+  private PutResult apply_ttl_update(NamedBlobRecord record, short accountId, short containerId, Connection connection)
+      throws Exception{
+    try (PreparedStatement statement = connection.prepareStatement(TTL_UPDATE_QUERY)) {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, record.getBlobName());
+      statement.setLong(4, record.getVersion());
+      statement.executeUpdate();
```

Comment:
You mean first check whether this record (..., name, version) exist in DB:
```
If yes: then apply the update.
Else: insert in a new row?
```

Reply:
I was thinking about if this record is gone, should we consider ttl update failed and throw exception like "this record does not exist"?

Reply:
Yeah, I can do that, @justinlin-linkedin how do you think?
I think maybe that's why initially I am just inserting a new row instead of updating the existing one. So that it doesn't need to check and then update.

Reply:
we should check out the return value of `executeUpdate`. It should not be 0. If the return value of `executeUpdate` is 0, then the record doesn't exist.

Reply:
Done, @SophieGuo410 @justinlin-linkedin can you please take another look?